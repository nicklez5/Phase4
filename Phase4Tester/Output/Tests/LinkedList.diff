--- SelfTestCases/LinkedList.out	2011-04-28 14:55:21.000000000 -0700
+++ ./Output/Tests/LinkedList.out	2020-03-14 21:50:40.326514317 -0700
@@ -1,27 +1,4 @@
-25
-10000000
-39
-25
-10000000
-22
-39
-25
-1
-0
-10000000
-28
-22
-39
-25
-2220000
--555
--555
-28
-22
-25
-33300000
-22
-25
-44440000
-0
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/LinkedList.compiler.out line 335 column 6: "$s8": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/LinkedList.compiler.out line 452 column 6: "$s8": operand is of incorrect type
 
+Processing terminated due to errors.
