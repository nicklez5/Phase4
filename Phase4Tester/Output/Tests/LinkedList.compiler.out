.data

vmt_Element:
  Element.Init
  Element.GetAge
  Element.GetSalary
  Element.GetMarried
  Element.Equal
  Element.Compare

vmt_List:
  List.Init
  List.InitNew
  List.Insert
  List.SetNext
  List.Delete
  List.Search
  List.GetEnd
  List.GetElem
  List.GetNext
  List.Print

vmt_LL:
  LL.Start

.text

  jal Main
  li $v0 10
  syscall

Main:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  li $a0 4
  jal _heapAlloc
  move $t0 $v0
  la $t9 vmt_LL
  sw $t9 0($t0)
  bnez $t0 null1
  la $a0 _str0
  j _error
null1:
  lw $t1 0($t0)
  lw $t1 0($t1)
  move $a0 $t0
  jalr $t1
  move $t1 $v0
  move $a0 $t1
  jal _print
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

Element.Init:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  move $t0 $a0
  move $t1 $a1
  move $t2 $a2
  move $t3 $a3
  move $t0 $v0
